### ISSUE TITLE
feat: Audio recorder

### ISSUE THREAD

**Issue by amaljosea** [chatwoot/chatwoot-mobile-app#757]
Same as in Chatwoot web it would be great if we could record audio in Chatwoot mobile.

We already have the backend system to make this work I guess.

![Screenshot 2023-12-31 at 9 58 31 PM](https://uploads.linear.app/2847f7b9-2caa-474c-b261-1c9bdc2a5da6/1e99b037-b17b-4d89-a60f-89d2b2602780/f10ba548-674d-46f3-a4c1-f17d5ab71608?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzI4NDdmN2I5LTJjYWEtNDc0Yy1iMjYxLTFjOWJkYzJhNWRhNi8xZTk5YjAzNy1iMTdiLTRkODktYTYwZi04OWQyYjI2MDI3ODAvZjEwYmE1NDgtNjc0ZC00NmYzLWE0YzEtZjE3ZDVhYjcxNjA4IiwiaWF0IjoxNzA0NDg1NTc2LCJleHAiOjMzMjc1MDQ1NTc2fQ.smLYTDeDq3ZH3wFXLDdgKB2ipX73oGX7GbnWIKPGkco "=200x")

**Comment by amaljosea** [comment]
@muhsin-k @nithindavid 

Can you please advise on the implementation? I think there are no blockers.

**Comment by Matheuskmrs** [comment]
Sabem me dizer se foi implementado essa função de reproduzir o áudio? Gostaria muito de adicionar no meu projeto.

**Comment by stgcompany** [comment]
Up!

**Comment by johnnysaito** [comment]
fala pessoal, alguma novidade sobre esse tópico, de gravar e ouvir o audio??

**Comment by johnnysaito** [comment]
up

**Comment by souzagaabriel** [comment]
This is a very decisive functionality for implementation across multiple clients.

**Comment by Matheuskmrs** [comment]
@sojan-official Hello! Do you have any plans regarding the implementation of the audio recorder and also audio playback on the chat screen?

**Comment by Midasstart** [comment]
How can we help launch these features?

**Comment by pGommes** [comment]
How can we help launch these features?

**Comment by muhsin-k** [comment]
Thanks for raising this issue. We are working on it.

**Comment by jjmmbb** [comment]
This is a MUST HAVE feature.

**Comment by hereck** [comment]
News about this implementation?

**Comment by souzagaabriel** [comment]
Any updates?

**Comment by lucianodltec** [comment]
+1

**Comment by viniciusccosta** [comment]
Is there a way to help ?

**Comment by deivisonrpg** [comment]
up

**Comment by admdanielspalma** [comment]
We do need it a lot!

**Comment by ViperTecCorporation** [comment]
up

**Comment by williamjduarte** [comment]
Please we need this feature.

**Comment by Matheuskmrs** [comment]
Is there a target date for adding this feature? @muhsin-k

**Comment by lucianodltec** [comment]
I would pay for this feature, let's start an online fundraiser.

**Comment by muhsin-k** [comment]
We have revamped the mobile app with a fresh look and added the option to play the audios. You can expect the recorder option in the upcoming releases.

Please try it out by joining our beta program.

**iOS**: https://testflight.apple.com/join/yQ4yoSx4

**Android**: https://play.google.com/apps/testing/com.chatwoot.app

Please report any issues you find in this thread: https://github.com/chatwoot/chatwoot-mobile-app/issues/862.

**Comment by williamjduarte** [comment]
@muhsin-k  It would be great to release the new version with the audio recording as well, please see if this is possible in January! I will include you in my prayers! 🥇

**Comment by muhsin-k** [comment]
We have added audio recording functionality. To try it out, join our beta program and install version 4.0.4 

**iOS**: https://testflight.apple.com/join/yQ4yoSx4

**Android**: https://play.google.com/apps/testing/com.chatwoot.app

Please report any issues you find here.

**Comment by dlc-letelier** [comment]
Just tested and it is working. Except, it is duplicating the audio that was send. It only appears duplicated in the IOS app and not in the Chatwoot itself (browser).

![Image](https://github.com/user-attachments/assets/f8825a70-44c3-4fb7-a76d-656c2070b65d)

**Comment by muhsin-k** [comment]
@dlc-letelier Thanks for testing this feature. Is it happening all the time?

**Comment by dlc-letelier** [comment]
@muhsin-k Yes, all the time. When I sent the message yesterday, I tried only once. But I've just send another voice message in the same conversation and the audio was duplicated again. It takes some seconds to the 'second audio' appear. Maybe, it has a relation when the messages arrives to the Chatwoot server.

**Comment by muhsin-k** [comment]
@dlc-letelier Thank you for testing. I have added strong validation to prevent duplicate audio messages. Please install the latest version 4.0.0 (3610) and let me know if you still experience the issue.

**Comment by muhsin-k** [comment]
@dlc-letelier It's so weird. Can you please add a screencast for this?

**Comment by dlc-letelier** [comment]
@muhsin-k Very weird. Here you are.


![Image](https://github.com/user-attachments/assets/cd4ae1a6-a652-4f91-ba5e-b8cd8ddf67dd)

https://github.com/user-attachments/assets/76ef3790-a581-480d-8473-d2964b57e34d

**Comment by muhsin-k** [comment]
@deivisonrpg Let's continue this discussion on [Discord](https://discord.com/channels/647412545203994635/1006470116432695368) If it is possible, as I might need to debug using your account.

**Comment by deivisonrpg** [comment]
> [@deivisonrpg](https://github.com/deivisonrpg) Let's continue this discussion on [Discord](https://discord.com/channels/647412545203994635/1006470116432695368) If it is possible, as I might need to debug using your account.

@muhsin-k  I think you tagged the wrong person 😀
